Automattic/jetpack
# Contact Form: refactor admin to use forms package

## Proposed changes:

As part of the Contact Form module deprecation, this PR refactors the `Grunion_Admin` and related callbacks to reference the Forms package.

The refactored callbacks call methods from the `Grunion_Admin` singleton. The `Grunion_Admin` singleton itself now references the `Admin` singleton. Hooks additions are removed from the `admin.php` file since they are added in the `Admin` constructor and we don't want to call the callbacks twice.

### Other information:


## Jetpack product discussion


Project thread: pf5801-Aj-p2
Issue: https://github.com/Automattic/jetpack/issues/35653

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






Let's do some regression testing.

- Spin up a test site with this branch
- Update the file `/projects/plugins/jetpack/modules/contact-form.php` so the condition on line 33 is falsy (or comment the block). This ensures we're loading the Contact Form module.
- Install the Classic Editor plugin and activate it
- Create a new post
- Check you can still add a Contact Form block and that nothing breaks
- Open the preview and verify you can see and use the form
